### PR TITLE
Fix malformat hash key handling.

### PR DESCRIPTION
**Pull request by qingling128**

### CODE REVIEW

File: test/plugin/base_test.rb [hunk 1]
```diff
@@ -544,6 +544,33 @@ def test_struct_payload_log
     end
   end
 
+  def test_struct_payload_malformat_log
```
Comment (igorpeshansky) [inline test/plugin/base_test.rb]: s/malformat/malformatted/. Also in the PR title.

File: test/plugin/base_test.rb [hunk 2]
```diff
@@ -544,6 +544,33 @@ def test_struct_payload_log
...
+    setup_gce_metadata_stubs
+    message = 'test message'
+    setup_logging_stubs do
+      d = create_driver
+      d.emit(
+        'int_as_key' => { 1 => message },
```
Comment (igorpeshansky) [inline test/plugin/base_test.rb]: I'd just use `'int_key'`, `'array_key'`, `'hash_key'`, `'nil_key'`. Also, you forgot symbol (e.g., `{ :symbol => message }`).

File: test/plugin/test_out_google_cloud_grpc.rb [hunk 3]
```diff
@@ -131,6 +131,26 @@ def test_severities
     end
   end
 
+  def test_struct_payload_non_utf8_log
+    setup_gce_metadata_stubs
+    setup_logging_stubs do
+      d = create_driver
+      d.emit('msg' => log_entry(0),
+             'normal_key' => 'test' + non_utf8_character + 'some non utf8',
```
Comment (igorpeshansky) [inline test/plugin/test_out_google_cloud_grpc.rb]: This is the test we should have added earlier but didn't, right?

File: test/plugin/test_out_google_cloud_grpc.rb [hunk 4]
```diff
@@ -131,6 +131,26 @@ def test_severities
...
+             'non_utf8' + non_utf8_character + 'key' => 5000,
+             'some_null_field' => nil)
```
Comment (igorpeshansky) [inline test/plugin/test_out_google_cloud_grpc.rb]: Do we also want to test the case where a non-utf8 character appears nested in a structured payload?

Comment (qingling128) [inline test/plugin/test_out_google_cloud_grpc.rb]: Added.

File: test/plugin/test_out_google_cloud_grpc.rb (hunk 3 above)
Comment (qingling128) [inline test/plugin/test_out_google_cloud_grpc.rb]: Yes. Should have added this.

File: test/plugin/base_test.rb (hunk 2 above)
Comment (qingling128) [inline test/plugin/base_test.rb]: Key names are changed and a test case for symbol is added. Seems `{ symbol: message }` is preferred in Ruby 1.9 though. (got Rubocop errors for `{ :symbol => message }`

File: test/plugin/base_test.rb (hunk 1 above)
Comment (qingling128) [inline test/plugin/base_test.rb]: Changed in this test. Will change the PR title when merging the PR.

**Comment by igorpeshansky** [comment]
LGTM :shipit:

File: lib/fluent/plugin/out_google_cloud.rb [hunk 5]
```diff
@@ -1044,7 +1044,7 @@ def list_from_ruby(arr)
     def struct_from_ruby(hash)
       ret = Google::Protobuf::Struct.new
       hash.each do |k, v|
-        ret.fields[k] ||= value_from_ruby(v)
+        ret.fields[convert_to_utf8(k.to_s)] ||= value_from_ruby(v)
```
Comment (igorpeshansky) [inline lib/fluent/plugin/out_google_cloud.rb]: Are we sure the `to_s` method will do the same thing as the REST path (which probably represents these as a JSON string instead)?

**Comment by igorpeshansky** [comment]
Looks like I was too late to add that last comment... Let's make a note that we should at least check it, but proceed with the release. Definitely doesn't block the `0.5.4.grpc.alpha.1` build.

**Pull request merged**